### DISCUSSION TITLE
Repository 'type': 'NOT_FOUND' graphQL API

### DISCUSSION THREAD

**Discussion by 1121589** [community/community#24737] [Enterprise]
<p>Hi,</p>
<p>Recently i’ve been working with graphql API on github and i have some querys working correctly, including a mutation.</p>
<p>This is on Enterprise btw.</p>
<p>One simple query that i use is the following (owner/name as an example):</p>
<pre><code class="lang-auto">query MyQuery {
  repository(name: "name", owner: "owner") {
    id
  }
}
</code></pre>
<p>But there is a weird problem.<br>
It works fine on my PC (Python Script with requests library) and a few others PC’s aswell.<br>
But on some PC’s, and most importantly on our build machine, it says NOT_FOUND on 2 out of 33 different repository’s (always the same 2).</p>
<p>The error is the following (owner/name as an example):</p>
<blockquote>
<p>{‘type’: ‘NOT_FOUND’, ‘path’: [‘repository’], ‘locations’: [{‘line’: 3, ‘column’: 9}], ‘message’: “Could not resolve to a Repository with the name ‘owner/name’.”}</p>
</blockquote>
<p>Some things that i have tried:</p>
<ul>
<li>Different tokens</li>
<li>Curl command seems to work, even on the affected PC’s.</li>
</ul>
<p>Line of code that i use for the request on Python.</p>
<pre><code class="lang-auto">response = requests.post(url=GRAPHQL_URL, json=query, headers=self._headers)
</code></pre>
<p>Does anyone had this problem aswell?</p>
<p>Thank you.</p>

**Comment by laurentS** [comment]
<p>Looks like the same issue as <a href="https://github.community/t/graphql-repository-query-fails-to-redirect-in-some-cases/184948" class="inline-onebox">Graphql repository query fails to redirect in some cases</a></p>

**Comment by 1121589** [comment, marked as answer]
<p>Hi,</p>
<p>Meantime i managed to solve this.</p>
<p>Not entirely sure about the science behind this.<br>
After printing the post payload that i was sending to github (<code>response.requests.headers</code>) i found out that the requests library was dropping the headers. After some reading it seems that it’s due to a redirection of url and for security reasons it drops the headers that contains my token.</p>
<p>What i did to solve was the following:<br>
(This class is used to avoid using .netrc config)</p>
<pre><code class="lang-auto">class NullAuth(requests.auth.AuthBase):
    def __call__(self, r):
        return r
</code></pre>
<p>And than passed it along:</p>
<p><code>response = requests.post(url=GRAPHQL_URL, json=query, headers=self._headers, auth=NullAuth())</code></p>
